### PR TITLE
SNOW-494740 Add Blocked as a Running Query

### PR DESCRIPTION
**Pull request by sfc-gh-jbahk**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-494740

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Adds blocked as a running query, in parallel with JDBC.

### CODE REVIEW

File: src/snowflake/connector/connection.py [hunk 1]
```diff
@@ -1435,9 +1435,10 @@ def is_still_running(status: QueryStatus) -> bool:
         """Checks whether given status is currently running."""
         return status in (
             QueryStatus.RUNNING,
-            QueryStatus.RESUMING_WAREHOUSE,
             QueryStatus.QUEUED,
+            QueryStatus.RESUMING_WAREHOUSE,
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/connection.py:1439]: Why did you move this line around?

**Review by sfc-gh-mkeller — APPROVED** [review]
Looks good, I just have the one question

File: src/snowflake/connector/connection.py (hunk 1 above)
Comment (sfc-gh-jbahk) [reply src/snowflake/connector/connection.py:1439]: In the enum in constants.py, RUNNING = 0, QUEUED = 5 & RESUMING_WAREHOUSE = 8, so made it align with the order. No big reason other than parity.
